presto-main/src/main/java/com/facebook/presto/sql/planner/plan/TopNRowNumberNode.java

```diff
@@ -66,6 +66,10 @@ public TopNRowNumberNode(
         this.maxRowCountPerPartition = maxRowCountPerPartition;
         this.partial = partial;
         this.hashSymbol = hashSymbol;
+
+        if (maxRowCountPerPartition < 0) {
```

Comment:
It looks like a hack. Can you explain why? I would prefer to fix the code which calls this constructor.

Reply:
I think to enforce `maxRowCountPerPartition > 0` is valid. Users should not write something like `row_number() = -1`. It's just like C++ compiler will warn the following code:
```c++
uint8_t n = ...;
if (n == -1) {
    ...
}

Reply:
`row_number() = -1` is definitely not very useful in a query, but looks like this should be valid SQL.
During planning, when we have proper context (e.g. when creating this node or at some earlier point), we may eliminate this, or add `Filter(false)` node above.

Reply:
> During planning, when we have proper context (e.g. when creating this node or at some earlier point), we may eliminate this, or add Filter(false) node above.

Exactly. This is a bug in the rule that creates the TopNRowNumber node. It should never create one if N <= 0